AST.fs

```diff
@@ -46,7 +46,7 @@ module Types =
     type Expression' =
         | True // true
         | False // false
-        | Int of int64 // 42
+        | Num of int64 // 42
```

Comment:
What was the reasoning behind this, btw?  Was keeping this as `Int` in case we ever added a floating-point literal.